jfxtras-agenda/src/main/java/jfxtras/internal/scene/control/skin/agenda/base24hour/AgendaSkinTimeScale24HourAbstract.java

```diff
@@ -238,7 +241,15 @@ public void refresh() {
 		refreshLocale();
 		setupAppointments();
 		nowUpdateRunnable.run(); 
-	}
+	}
+
+	/**
+	 * 
+	 */
+	@Override
+    public Pane getNodeForPopup(Appointment appointment) { return appointmentNodeMap.get(appointment); }
+    final private Map<Appointment, Pane> appointmentNodeMap = new WeakHashMap<>();
```

Comment:
1. Why do you need equals for unit testing? I've written agenda and a dozen other controls without it. You could write a static SomeUtil.isEqualTo(o1, o2). That said, Appointment is just an interface, so a user may provide his own implementation with equals en hashCode, so we cannot rely on these assumptions. The question would then be: why use a hashcode based map? 
But then equals quickly becomes a problem. Maybe the best solution would be to use System.	identityHashCode(u) as the key, not a weak map and use a deletion listener on appointments to clean up the map.

2. not an option

3. I do not think a few simple loops will hurt that much in the grand scheme of things. But as said, you cannot use equals because there may two appointments with the same content, which we do not consider being the same.

We either decide to fully base ourselves on System.identityHashCode(u) or tell the user not to implement equals and hashcode.